### ISSUE TITLE
Many selectable fonts are not actually available (OSX)

### ISSUE THREAD

**Issue by jemus42** [calligross/ggthemeassist#35]
Error when selecting an arbitary non-installed font:

```
Warning in grid.Call(L_textBounds, as.graphicsAnnot(x$label), x$x, x$y,  :
  no font could be found for family "Japan1"
Warning in grid.Call(L_textBounds, as.graphicsAnnot(x$label), x$x, x$y,  :
  no font could be found for family "Japan1"
Warning: Error in grid.Call: polygon edge not found
Stack trace (innermost first):
    110: grid.Call
    109: heightDetails.text
    108: heightDetails
    107: <Anonymous>
    106: grid.Call.graphics
    105: push.vp.viewport
    104: push.vp
    103: push.vp.vpTree
    102: FUN
    101: lapply
    100: pushViewport
     99: pushgrobvp.viewport
     98: pushgrobvp
     97: pushvpgp
     96: preDraw.gTree
     95: <Anonymous>
     94: grid.Call.graphics
     93: push.vp.viewport
     92: FUN
     91: lapply
     90: pushViewport
     89: pushgrobvp.viewport
     88: pushgrobvp
     87: pushvpgp
     86: preDraw.gTree
     85: preDraw
     84: drawGTree
     83: recordGraphics
     82: grid.draw.gTree
     81: grid.draw
     80: print.ggplot
     79: print
     78: renderPlot
     70: output$ThePlot4
      3: shiny::runApp
      2: runGadget
      1: ggThemeAssist:::ggThemeAssist
```

Possible solutions: Maybe check which fonts are locally available on startup and use those? I don't know, fonts are a pain.  
I recently used the extrafont package and `font_import` so import ***all*** the fonts on my system, so I thought I'm good to go font-wise, but alas, it was for naught. 

System info:

```
Session info --------------------------------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.2.4 (2016-03-10)
 system   x86_64, darwin13.4.0        
 ui       RStudio (0.99.878)          
 language (EN)                        
 collate  en_US.UTF-8                 
 tz       Europe/Berlin               
 date     2016-03-12                  

Packages ------------------------------------------------------------------------------------------------------------
 package       * version     date       source                                   
 acepack         1.3-3.3     2013-05-03 CRAN (R 3.1.0)                           
 addinexamples   0.1.0       2016-01-19 Github (rstudio/addinexamples@646777d)   
 ArgumentCheck   0.10.0      2015-07-10 CRAN (R 3.2.0)                           
 assertthat      0.1         2013-12-06 CRAN (R 3.1.0)                           
 broom         * 0.4.0       2015-11-30 CRAN (R 3.2.3)                           
 car             2.1-1       2015-12-14 CRAN (R 3.2.3)                           
 cluster         2.0.3       2015-07-21 CRAN (R 3.2.0)                           
 codetools       0.2-14      2015-07-15 CRAN (R 3.2.0)                           
 coin            1.1-2       2015-11-16 CRAN (R 3.2.2)                           
 colorspace      1.2-6       2015-03-11 CRAN (R 3.2.0)                           
 cowplot         0.6.1       2016-03-06 CRAN (R 3.2.4)                           
 DBI             0.3.1       2014-09-24 CRAN (R 3.1.1)                           
 ddateR          0.3         2016-02-18 local                                    
 devtools        1.10.0      2016-01-23 CRAN (R 3.2.3)                           
 digest          0.6.9       2016-01-08 CRAN (R 3.2.3)                           
 dplyr         * 0.4.3       2015-09-01 CRAN (R 3.2.0)                           
 DT            * 0.1         2015-06-09 CRAN (R 3.2.0)                           
 DYM             0.2         2016-01-22 CRAN (R 3.2.3)                           
 effects         3.0-7       2016-02-20 CRAN (R 3.2.3)                           
 extrafont     * 0.17        2014-12-08 CRAN (R 3.2.0)                           
 extrafontdb     1.0         2012-06-11 CRAN (R 3.2.0)                           
 foreign         0.8-66      2015-08-19 CRAN (R 3.2.0)                           
 formatR         1.3         2016-03-05 CRAN (R 3.2.4)                           
 formattable   * 0.1.5       2015-12-03 CRAN (R 3.2.3)                           
 Formula         1.2-1       2015-04-07 CRAN (R 3.2.0)                           
 fortunes        1.5-2       2013-12-14 CRAN (R 3.1.0)                           
 ggplot2       * 2.1.0       2016-03-01 CRAN (R 3.2.3)                           
 ggThemeAssist   0.1.9000    2016-03-10 Github (calligross/ggthemeassist@faa0588)
 gridExtra       2.2.1       2016-02-29 CRAN (R 3.2.4)                           
 gtable          0.2.0       2016-02-26 CRAN (R 3.2.3)                           
 haven           0.2.0       2015-04-09 CRAN (R 3.2.0)                           
 Hmisc           3.17-2      2016-02-21 CRAN (R 3.2.3)                           
 htmltools       0.3         2015-12-29 CRAN (R 3.2.3)                           
 htmlwidgets     0.6         2016-02-25 CRAN (R 3.2.3)                           
 httpuv          1.3.3       2015-08-04 CRAN (R 3.2.0)                           
 jsonlite        0.9.19      2015-11-28 CRAN (R 3.2.2)                           
 knitr           1.12.3      2016-01-22 CRAN (R 3.2.3)                           
 lattice         0.20-33     2015-07-14 CRAN (R 3.2.0)                           
 latticeExtra    0.6-28      2016-02-09 CRAN (R 3.2.3)                           
 lazyeval        0.1.10      2015-01-02 CRAN (R 3.1.2)                           
 lazyWeave       3.0.1       2016-01-09 CRAN (R 3.2.3)                           
 lme4            1.1-11      2016-02-12 CRAN (R 3.2.3)                           
 lmtest          0.9-34      2015-06-06 CRAN (R 3.2.0)                           
 lubridate     * 1.5.0       2015-12-03 CRAN (R 3.2.3)                           
 magrittr      * 1.5         2014-11-22 CRAN (R 3.2.0)                           
 markdown        0.7.7       2015-04-22 CRAN (R 3.2.0)                           
 MASS            7.3-45      2015-11-10 CRAN (R 3.2.2)                           
 Matrix          1.2-4       2016-03-02 CRAN (R 3.2.4)                           
 MatrixModels    0.4-1       2015-08-22 CRAN (R 3.2.0)                           
 memoise         1.0.0       2016-01-29 CRAN (R 3.2.3)                           
 mgcv            1.8-12      2016-03-03 CRAN (R 3.2.4)                           
 mime            0.4         2015-09-03 CRAN (R 3.2.0)                           
 miniUI          0.1.1       2016-01-15 CRAN (R 3.2.3)                           
 minqa           1.2.4       2014-10-09 CRAN (R 3.1.2)                           
 mnormt          1.5-4       2016-03-09 CRAN (R 3.2.4)                           
 modeltools      0.2-21      2013-09-02 CRAN (R 3.2.0)                           
 multcomp        1.4-4       2016-02-17 CRAN (R 3.2.3)                           
 munsell         0.4.3       2016-02-13 CRAN (R 3.2.3)                           
 mvtnorm         1.0-5       2016-02-02 CRAN (R 3.2.3)                           
 nlme            3.1-125     2016-02-27 CRAN (R 3.2.3)                           
 nloptr          1.0.4       2014-08-04 CRAN (R 3.1.2)                           
 nnet            7.3-12      2016-02-02 CRAN (R 3.2.3)                           
 pbkrtest        0.4-6       2016-01-27 CRAN (R 3.2.3)                           
 pixiedust     * 0.6.1       2016-01-23 CRAN (R 3.2.3)                           
 plyr            1.8.3       2015-06-12 CRAN (R 3.2.0)                           
 psych           1.5.8       2015-08-30 CRAN (R 3.2.0)                           
 purrr         * 0.2.1       2016-02-13 CRAN (R 3.2.3)                           
 quantreg        5.21        2016-02-13 CRAN (R 3.2.3)                           
 R6              2.1.2       2016-01-26 CRAN (R 3.2.3)                           
 RColorBrewer    1.1-2       2014-12-07 CRAN (R 3.1.2)                           
 Rcpp            0.12.3      2016-01-10 CRAN (R 3.2.3)                           
 reshape2        1.4.1       2014-12-06 CRAN (R 3.1.2)                           
 rmarkdown       0.9.5       2016-02-19 Github (rstudio/rmarkdown@8b8ce77)       
 rpart           4.1-10      2015-06-29 CRAN (R 3.2.0)                           
 rsconnect       0.4.1.11    2016-02-19 Github (rstudio/rsconnect@2419667)       
 rstudioapi      0.5         2016-01-24 CRAN (R 3.2.3)                           
 Rttf2pt1        1.3.3       2015-01-13 CRAN (R 3.2.0)                           
 ryouready       0.4         2015-12-03 CRAN (R 3.2.3)                           
 sandwich        2.3-4       2015-09-24 CRAN (R 3.2.0)                           
 scales        * 0.4.0       2016-02-26 CRAN (R 3.2.3)                           
 shiny         * 0.13.1      2016-02-17 CRAN (R 3.2.3)                           
 sjmisc          1.5         2016-02-11 CRAN (R 3.2.3)                           
 sjPlot        * 1.9.2       2016-03-02 CRAN (R 3.2.3)                           
 SparseM         1.7         2015-08-15 CRAN (R 3.2.0)                           
 stringdist      0.9.4.1     2016-01-02 CRAN (R 3.2.3)                           
 stringi         1.0-1       2015-10-22 CRAN (R 3.2.0)                           
 stringr         1.0.0       2015-04-30 CRAN (R 3.2.0)                           
 survival        2.38-3      2015-07-02 CRAN (R 3.2.0)                           
 tadaatoolbox  * 0.10.0.9000 2016-03-08 local                                    
 TH.data         1.0-7       2016-01-28 CRAN (R 3.2.3)                           
 tidyr           0.4.1       2016-02-05 CRAN (R 3.2.3)                           
 vcd             1.4-1       2015-07-08 CRAN (R 3.2.0)                           
 viridis       * 0.3.3       2016-03-04 CRAN (R 3.2.4)                           
 waffle        * 0.5.1       2016-02-01 Github (hrbrmstr/waffle@e298c54)         
 XML             3.98-1.4    2016-03-01 CRAN (R 3.2.3)                           
 xtable          1.8-2       2016-02-05 CRAN (R 3.2.3)                           
 zoo             1.7-12      2015-03-16 CRAN (R 3.2.0) 
````

**Comment by jemus42** [comment]
Update: According to @MarauderPixie, the font situation on Ubuntu and Windows is similarly unfortunate

**Comment by calligross** [comment]
We will look into it, we need a function which returns the actual available fonts.

At the moment it's a quick & dirty hack: `names(pdfFonts())`

**Comment by jemus42** [comment]
Could `extrafont::fonttable()$FamilyName` help?  
It spits out >200 fonts for me, but that's probably only the case since I imported a bunch recently.

**Comment by calligross** [comment]
Should be fixed by 824859b.

**Comment by jemus42** [comment]
Looks better on my end, but I'd suggest leaving this issue open in case this doesn't work for people who didn't recently did the extrafont dance.
